omod/src/main/java/org/openmrs/module/idgen/rest/resource/AutoGenerationOptionResource.java

```diff
@@ -0,0 +1,100 @@
+        "1.9.*", "1.10.*", "1.11.*", "1.12.*", "2.0.*", "2.1.*" })
+public class AutoGenerationOptionResource extends DelegatingCrudResource<AutoGenerationOption> {
+  @Override
+	public DelegatingResourceDescription getRepresentationDescription(Representation rep) {
+		DelegatingResourceDescription description = null;
+		
+		// NOTE: description for ref representation is provided by MetaDataDelegatingCrudResource
+		if (rep instanceof DefaultRepresentation || rep instanceof FullRepresentation) {
+      description = new DelegatingResourceDescription();
+      
+			description.addProperty("identifierType");
+			description.addProperty("location");
+			description.addProperty("source");
+			description.addProperty("manualEntryEnabled");
+			description.addProperty("automaticGenerationEnabled");
+		}
+		
+		return description;
+  }
+  
+  @Override
+  public DelegatingResourceDescription getCreatableProperties() {
+      DelegatingResourceDescription description = new DelegatingResourceDescription();
+      description.addProperty("identifierType");
+      description.addProperty("location");
+			description.addProperty("source");
+			description.addProperty("manualEntryEnabled");
+			description.addProperty("automaticGenerationEnabled");
+      return description;
+  }
+
+  @Override
+  public DelegatingResourceDescription getUpdatableProperties() {
+      DelegatingResourceDescription description = new DelegatingResourceDescription();
+      description.addProperty("location");
+			description.addProperty("source");
+			description.addProperty("manualEntryEnabled");
+			description.addProperty("automaticGenerationEnabled");
+      return description;
+  }
+
+  @Override
+  public AutoGenerationOption newDelegate() {
+    return new AutoGenerationOption();
+  }
+
+  @Override
+  public AutoGenerationOption save(AutoGenerationOption autoGenerationOption) {
+    return Context.getService(IdentifierSourceService.class).saveAutoGenerationOption(autoGenerationOption);
+  }
+
+  @Override
+  public AutoGenerationOption getByUniqueId(String uniqueId) {
+    return Context.getService(IdentifierSourceService.class).getAutoGenerationOption(Integer.parseInt(uniqueId));
+  }
+
+  @Override
+  public void purge(AutoGenerationOption delegate, RequestContext context) throws ResponseException {
+    Context.getService(IdentifierSourceService.class).purgeAutoGenerationOption(delegate);
+  }
+
+  @Override
+  public void delete(AutoGenerationOption delegate, String id, RequestContext context) throws ResponseException {
+    throw new ResourceDoesNotSupportOperationException("Deleting of AutoGenerationOption is not supported");
+  }
+
+  @Override
+  protected PageableResult doGetAll(RequestContext context) throws ResponseException {
+		return new AlreadyPaged<AutoGenerationOption>(Context.getService(IdentifierSourceService.class).getAutoGenerationOption(3), context);
```

Comment:
What does the 3 parameter mean?

Reply:
I was trying to implement the `doGetAll` method and I noticed that from all the other examples that I checked, it only accepts one parameter which is the `context`. `getAutoGenerationOption` also requires a parameter and I don't know how to pass that in. That's why I hardcode the `3` just for testing purposes.

Reply:
Did you take a look at how that list is generated in AutoGenerationOptionController?

Reply:
Yes I did. I noticed they first get all the patientIdentifierTypes and they did a for loop to get all AutoGenerationOptions for each of them. I tried to replicate the same thing but it doesn't work.

Reply:
What do you mean by it does not work?

Reply:
The return line get's highlighted complaining that `the constructor AlreadyPaged<AutoGenerationOption>(Map<PatientIdentifierType,List<AutoGenerationOption>>, RequestContext) is undefined`. Here is the snapshot https://snag.gy/hWHMmk.jpg

Reply:
Do not pass in a map, instead pass in a list of options.

Reply:
I'm still having the same error even after changing the options to list.

Reply:
Can you commit those changes and i take a look?

Reply:
Done!